packages/core/src/common/preferences/preference-service.ts

```diff
@@ -363,6 +363,7 @@ export class PreferenceServiceImpl implements PreferenceService {
                     const scopeValue: JSONValue | undefined = provider.get(preferenceName);
                     // Skip if a more specific scope has this preference defined
                     if (scope > change.scope && scopeValue !== undefined) {
+                        acceptChange(change);
```

Comment:
> @colin-grant-work if this is not a good solution, then how to fix the main issue above?

Rather than deleting this and allow changes through unconditionally - i.e. always firing events, even when the value to be exposed externally hasn't changed - check the schema of the affected preference. If it's an object (or array, if those should also be merged), perform the merge and then fire the event. If it's a primitive, then we still shouldn't fire the change.

There should be new tests for this, but the existing tests that check that we don't fire more events than necessary for primitive-type preferences shouldn't be affected.